src/function.ts

```diff
@@ -30,11 +38,50 @@ export type AnyLambda = Function<any, any>;
  */
 export class Function<P, O> {
   readonly kind = "Function" as const;
+  public static readonly FunctionlessType = "Function";
+  readonly resource: aws_lambda.IFunction;
+
+  static promises: Promise<any>[] = [];
 
   // @ts-ignore - this makes `F` easily available at compile time
   readonly __functionBrand: ConditionalFunction<P, O>;
 
-  constructor(readonly resource: aws_lambda.IFunction) {
+  constructor(
```

Comment:
We should add thorough code documentation with examples for each variant.

Reply:
In a future PR, we should import for the existing case instead of using the constructor.

```ts
new Function(this, id, async () => {})

Function.fromFunction(new aws_lambda.Function(...));
```